### PR TITLE
USWDS-3358 Time Picker

### LINKED ISSUE #3358: Create a time picker component.
## User Story

As a front-end developer, I need a way to help users select a time in a way that is compact, functional, and accessible.

## Examples
* [USPTO](https://uspto.github.io/designpatterns/1.x/docs/components/formControls.html#time-picker)
* [Lightning](https://www.lightningdesignsystem.com/components/timepicker/)

## Reference
* [Designing The Perfect Date And Time Picker - Vitaly Friedman - 07/05/2017](https://www.smashingmagazine.com/2017/07/designing-perfect-date-time-picker/)

## Checklist
- [ ] Create low-fidelity design
- [ ] Review & discuss low-fi design (designer + developers)
- [ ] Get user feedback on low-fi designs (optional - if team feels is needed)
- [ ] Code
- [ ] Conduct usability test with working code (optional - if team feels is needed)
- [ ] Create hi-fi design assets (Adobe XD, Sketch, Figma)
- [ ] Review & approve hi-fi design assets
- [ ] Publish design assets (update downloadable files for Adobe XD, Sketch, Figma)

### PR DESCRIPTION
**Pull request by wbyoko**
closes #3358

## Description

Updates the combo box component and implements a time picker component that implements a combo box.

## Additional information

Screenshot 2020-07-28 13 24 58
Screenshot 2020-07-28 13 25 07
Screenshot 2020-07-28 13 25 16

Include any of the following (as necessary): 

* Relevant research and support documents
* Type of content review needed: stylistic or copy editing
* Screenshot images
* Notes

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Comment by thisisdano** [comment]
@Ben-Judy-Flexion Do you think this component should have a narrower default width because its content is relatively short?

**Comment by Ben-Judy-Flexion** [comment]
> @Ben-Judy-Flexion Do you think this component should have a narrower default width because its content is relatively short?

Yes, absolutely. Should default to just enough width to accommodate the longest timestamp. (Probably testing for width of 12:30 would do the trick, I think.)

**Comment by wbyoko** [comment]
@thisisdano @Ben-Judy-Flexion  

adding styles for this 

`.usa-time-picker {
  width: 10rem;
}
`

leads to this 

Screenshot 2020-07-28 13 18 55
Screenshot 2020-07-28 13 19 02
Screenshot 2020-07-28 13 19 16
Screenshot 2020-07-28 13 19 27


Is this okay? I can push this if so.

File: src/js/components/time-picker.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+const MAX_TIME = 60 * 24 - 1;
+const MIN_TIME = 0;
+const DEFAULT_STEP = 30;
+const MIN_STEP = 1;
+
+const FILTER_DATASET = {
+  filter:
+    "0?{{ hourQueryFilter }}:{{minuteQueryFilter}}.*{{ apQueryFilter }}m?",
+  apQueryFilter: "([ap])",
+  hourQueryFilter: "([1-9][0-2]?)",
```
Comment (kkoskelin) [inline src/js/components/time-picker.js:17]: I confess I don't know if this messes with your capture groups, but:
```
  const VALID_HOUR_RE = /^((0?[1-9])|([1][0-2]))$/; // 1-9, 01-09, 10-12
  const VALID_MINUTE_RE = /^([0-5][0-9])$/; // 00-59
```

Comment (wbyoko) [reply src/js/components/time-picker.js:17]: At the end of the day these are not validating regular expressions; instead these are query filters to help build a dynamic regular expression that will then try to match against a combo box option.

So even though the regular expression query filter captures a lot it wouldn't match against anything in reality. 

Also, I believe that the one digit minute is needed for (04:1p) matching to (04:15pm)

Comment (kkoskelin) [reply src/js/components/time-picker.js:17]: That makes sense.  Thanks for the insight!

**Comment by katieekline** [comment]
@thisisdano The [copy](https://docs.google.com/document/d/12HiNYhkNQE5qkT_VTmYQJlwYP4mccY1VhBf6OZHkDkA/edit?usp=sharing) has been reviewed and is ready for final approval.

**Review by thisisdano — APPROVED** [review]

**Review by sarnikbixal — COMMENTED** [review]
I can't find approve button but looks good

**Pull request merged**